src/python/pants/backend/java/lint/google_java_format/rules.py

```diff
@@ -0,0 +1,173 @@
+    )
+
+    maybe_java16_or_higher_options = []
+    if jdk_setup.jre_major_version >= 16:
+        maybe_java16_or_higher_options = [
+            "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+        ]
+
+    maybe_aosp_option = []
+    if tool.aosp:
+        maybe_aosp_option = ["--aosp"]
+
+    args = [
+        *jdk_setup.args(bash, tool_classpath.classpath_entries()),
+        *maybe_java16_or_higher_options,
+        "com.google.googlejavaformat.java.Main",
+        *maybe_aosp_option,
+        "--dry-run" if setup_request.check_only else "--replace",
+        *source_files.files,
+    ]
+
+    process = Process(
+        argv=args,
+        input_digest=input_digest,
+        output_files=source_files_snapshot.files,
+        append_only_caches=jdk_setup.append_only_caches,
+        env=jdk_setup.env,
+        description=f"Run Google Java Format against {pluralize(len(setup_request.request.field_sets), 'file')}.",
+        level=LogLevel.DEBUG,
+    )
+
+    return Setup(process, original_digest=source_files_snapshot.digest)
+
+
+@rule(desc="Format with Google Java Format", level=LogLevel.DEBUG)
+async def google_java_format_fmt(
+    field_sets: GoogleJavaFormatRequest, tool: GoogleJavaFormatSubsystem
+) -> FmtResult:
+    if tool.skip:
+        return FmtResult.skip(formatter_name="Google Java Format")
+    setup = await Get(Setup, SetupRequest(field_sets, check_only=False))
+    result = await Get(ProcessResult, Process, setup.process)
+    return FmtResult.from_process_result(
+        result,
+        original_digest=setup.original_digest,
+        formatter_name="Google Java Format",
+        strip_chroot_path=True,
+    )
+
+
+@rule(desc="Lint with Google Java Format", level=LogLevel.DEBUG)
+async def google_java_format_lint(
+    field_sets: GoogleJavaFormatRequest, tool: GoogleJavaFormatSubsystem
+) -> LintResults:
+    if tool.skip:
+        return LintResults([], linter_name="Google Java Format")
+    setup = await Get(Setup, SetupRequest(field_sets, check_only=True))
+    result = await Get(FallibleProcessResult, Process, setup.process)
+    lint_result = LintResult.from_fallible_process_result(result)
+    if lint_result.exit_code == 0 and lint_result.stdout.strip() != "":
+        # Note: The formetter returns success even if it would have reformatted the files.
+        # When this occurs, convert the LintResult into a failure.
+        lint_result = dataclasses.replace(
+            lint_result,
+            exit_code=1,
+            stdout=f"The following Java files require formatting:\n{lint_result.stdout}\n",
```

Comment:
Maybe this to be less verbose?

```suggestion
            stdout=f"These files require formatting:\n{lint_result.stdout}\n",
```